### PR TITLE
Verify TEE tasks `enclaveSignature` before accepting IPFS upload

### PR DESCRIPTION
**Pull request by jbern0rd**

### CODE REVIEW

File: src/main/java/com/iexec/resultproxy/proxy/ProxyService.java [hunk 1]
```diff
@@ -47,48 +49,61 @@
...
         }
+
+        // TEE tasks with ResultModel containing the enclave signature
+        if (!BytesUtils.EMPTY_ADDRESS.equals(model.getEnclaveSignature())) {
+            return authorizationService.checkEnclaveSignature(model, walletAddress);
+        }
+
+        // TODO remove this case in the future. As we support 2 stack versions, it will be a major after deprecated proxy controller endpoints removal
+        // TEE tasks with ResultModel without the enclave signature
+        boolean isActive = chainTask.getStatus() == ChainTaskStatus.ACTIVE;
```
Comment (thewhitewizard) [inline src/main/java/com/iexec/resultproxy/proxy/ProxyService.java:104]: we can pass these variables to final

File: src/main/java/com/iexec/resultproxy/proxy/ProxyService.java [hunk 2]
```diff
@@ -47,48 +49,61 @@
...
 
+        // TODO [PoCo Boost] on-chain deal id available in result model to avoid fetching task
+        ChainTask chainTask = iexecHubService.getChainTask(chainTaskId).orElse(null);
+        if (chainTask == null) {
+            log.error("Trying to upload result for TEE but getChainTask failed [chainTaskId:{}, uploader:{}]",
+                    chainTaskId, walletAddress);
+            return false;
+        }
+
+        ChainDeal chainDeal = iexecHubService.getChainDeal(chainTask.getDealid()).orElse(null);
```
Comment (thewhitewizard) [inline src/main/java/com/iexec/resultproxy/proxy/ProxyService.java:83]: we can pass these variables to final

File: src/main/java/com/iexec/resultproxy/proxy/ProxyService.java [hunk 3]
```diff
@@ -47,48 +49,61 @@
...
-            }
+        // check if result has been already uploaded
+        if (isResultFound(chainTaskId)) {
+            log.error("Trying to upload result that has been already uploaded [chainTaskId:{}, uploadRequester:{}]",
+                    chainTaskId, walletAddress);
+            return false;
+        }
 
+        // TODO [PoCo Boost] on-chain deal id available in result model to avoid fetching task
+        ChainTask chainTask = iexecHubService.getChainTask(chainTaskId).orElse(null);
```
Comment (thewhitewizard) [inline src/main/java/com/iexec/resultproxy/proxy/ProxyService.java:76]: we can pass these variables to final

File: src/main/java/com/iexec/resultproxy/authorization/AuthorizationService.java [hunk 4]
```diff
@@ -109,4 +114,34 @@ public String getChallengeForWorker(WorkerpoolAuthorization workerpoolAuthorizat
                 workerpoolAuthorization.getChainTaskId(),
                 workerpoolAuthorization.getEnclaveChallenge());
     }
+
+    // region workerpool authorization cache
+    public boolean checkEnclaveSignature(ResultModel model, String walletAddress) {
+        final String chainTaskId = model.getChainTaskId();
+        final String wpAuthKey = String.join("-", chainTaskId, walletAddress);
+        String resultHash = HashUtils.concatenateAndHash(chainTaskId, model.getDeterministHash());
```
Comment (thewhitewizard) [inline src/main/java/com/iexec/resultproxy/authorization/AuthorizationService.java:122]: several variables should be able to pass as final in this method

Comment (jbern0rd) [reply src/main/java/com/iexec/resultproxy/authorization/AuthorizationService.java]: done

File: src/main/java/com/iexec/resultproxy/proxy/ProxyService.java (hunk 3 above)
Comment (jbern0rd) [reply src/main/java/com/iexec/resultproxy/proxy/ProxyService.java]: done

**Review by thewhitewizard — APPROVED** [review]

**Pull request merged**